### ISSUE TITLE
Port twisted.internet.test.test_udp and test_udp_internals to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6023]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6023|
|Type|enhancement|
|Created|2012-09-27 18:17:25Z|
|Branch|https://github.com/twisted/twisted/tree/testudp-py3-6023|

Unfortunately `twisted.internet.test.test_udp` and `test_udp_internals` don't provide enough coverage to prove UDP works (that'll need to wait for `twisted.test.test_udp` which depends on `TestCase`), but they should be ported.

<details><summary>Searchable metadata</summary>

```
trac-id__6023 6023
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_testudp_py3_6023 branches/testudp-py3-6023
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1348769845000000 1348769845000000
changetime__1348859136000000 1348859136000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [35826]) Branching to 'testudp-py3-6023'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst set owner to @exarkun|
|-|-|

Start build: 

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/testudp-py3-6023

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

  1. `__future__` imports missing here and there
  1. Would be nice to have a ticket link in the comment in `test_unregisterProducerAfterDisconnect` in `connectionmixins.py`

Otherwise looks good, merge when the above are addressed.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set status to `closed`|
|-|-|

(In [35838]) Merge testudp-py3-6023.

Author: itamar
Review: exarkun
Fixes: [#6023](https://github.com/twisted/twisted/issues/6023)

Port twisted.internet.test.test_udp and test_udp_internals to Python 3.
